lualib/lua_selectors/extractors.lua

```diff
@@ -388,6 +388,54 @@ The first argument must be header name.]],
   ]],
     ['args_schema'] = { ts.one_of { 'stem', 'raw', 'norm', 'full' }:is_optional()},
   },
+  -- Get queue ID
+  ['queueid'] = {
+    ['get_value'] = function(task)
+      local queueid = task:get_queue_id()
+      if queueid then return queueid,'string' end
+      return nil
+    end,
+    ['description'] = [[Get queue ID]],
+  },
+  -- Get ID of the task being processed
+  ['uid'] = {
+    ['get_value'] = function(task)
+      local uid = task:get_uid()
+      if uid then return uid,'string' end
+      return nil
+    end,
+    ['description'] = [[Get ID of the task being processed]],
+  },
+  -- Get message ID of the task being processed
+  ['messageid'] = {
+    ['get_value'] = function(task)
+      local mid = task:get_message_id()
+      if mid then return mid,'string' end
+      return nil
+    end,
+    ['description'] = [[Get message ID]],
+  },
+  -- Get specific symbol
+  ['symbol'] = {
+    ['get_value'] = function(task, args)
+      local symbol = task:get_symbol(args[1])
```

Comment:
I'd suggest to simplify this code. In fact `get_symbol` always return a table with one element (legacy compatibility). So we can get it and greately simplify the former stuff by just returning a `table`. So `args[2]` will be redundant, since a table can be indexed by a table methods + transforms.

Reply:
Sounds perfect, thanks! The table_list gave me a hard time.
Returning the first element as a table, will make things a lot easier.
I will try it out as soon as i will be back at a pc (probably tommorrow).